docs/site/migration/auth/authentication.md

```diff
@@ -0,0 +1,296 @@
+```
+
+Then create the corresponding repositories for `User` and `UserCredentials`:
+
+```sh
+cd myLB4App
+$ lb4 repository
+? Please select the datasource DbDatasource
+? Select the model(s) you want to generate a repository User, UserCredentials
+? Please select the repository base class DefaultCrudRepository (Legacy juggler bridge)
+```
+
+Create the user controller for CRUD operations:
+
+```sh
+$ lb4 controller
+? Controller class name: User
+Controller User will be created in src/controllers/user.controller.ts
+
+? What kind of controller would you like to generate? REST Controller with CRUD functions
+? What is the name of the model to use with this CRUD repository? User
+? What is the name of your CRUD repository? UserRepository
+? What is the name of ID property? id
+? What is the type of your ID? number
+? Is the id omitted when creating a new instance? No
+? What is the base HTTP path name of the CRUD operations? /users
+   create src/controllers/user.controller.ts
+   update src/controllers/index.ts
+
+Controller User was created in src/controllers/
+```
+
+### Creating Login Function
+
+_Migrating all functions for User need more effort, this section only creates
+login function as the minimum requirement._
+
+For token based authentication, a user logs in by providing correct credentials
+(email and password) in the payload of endpoint `User/login`, then gets a token
+back with its identity information encoded and includes the token in the header
+of next requests.
+
+Let's add a new controller function `login` decorated with the REST decorators
+that describe the request and response (see
+[the complete user controller file](https://github.com/strongloop/loopback-next/tree/master/examples/access-control-migration/src/controllers/user.controller.ts)).
+
+The core logic of `login` does three things:
+
+- verifies the credentials and find the right user.
+  - calls `userService.verifyCredentials()` to do it.
+- converts the user into a standard principal shared across the authentication
+  and authorization modules.
+  - calls `userService.convertToUserProfile()` to do it.
+- encodes the principal that carries the user's identity information into a JSON
+  web token.
+  - calls `jwtService.generateToken()` to do it.
+
+As you could see, the login function is pretty light, all the complex utilities
+are implemented by services. This also decouples the responsibilities. We will
+create the token and user service in the next section.
+
+### Mounting Authentication Component
+
+The authentication component is temporarily kept in the
+[example repository](https://github.com/strongloop/loopback-next/tree/master/examples/access-control-migration/src/components/jwt-authentication).
+It includes all the services that verify a user's identity and manage the tokens
+and users. We plan to extract this prototype component into a standalone package
+in story [to be created](). Now you can import it from the example as a
```

Comment:
`story [to be created]()` don't forget to update this 😄

Reply:
created :)